### ISSUE TITLE
Should keep upgrading packages if one doesn't compile

### ISSUE THREAD

**Issue by Michael-F-Bryan** [nabijaczleweli/cargo-update#18]
I recently did a `cargo install-update --all` and in this case `clippy-lints` failed to compile. It looks like `cargo-update` exits on the first compile error because afterwards running `cargo install-update --list --all` shows that there are still packages with updates available. 

I had to manually update the others because `cargo-update` can't get past `clippy-lints`. Can you set it up so that instead of exiting on the first compilation error, `cargo-update` just skips that package and moves onto the next? Then at the end you could print a message saying something like "5 packages updated. The following package(s) failed to compile: cargo-clippy" to let the user know what happened.

**Comment by nabijaczleweli** [comment]
Yeah, sure

**Comment by nabijaczleweli** [comment]
New error system:

```
P:\Rust\cargo-update>target\debug\cargo install-update -a
    Updating registry `https://github.com/rust-lang/crates.io-index`

Package         Installed  Latest    Needs update
cargo-count     v0.2.2     v0.2.2    No
cargo-graph     v0.3.0     v0.3.0    No
cargo-outdated  v0.2.0     v0.2.0    No
cargo-update    v0.5.3     v0.5.3    No
checksums       v0.5.2     v0.5.2    No
clippy          v0.0.1     v0.0.103  Yes
identicon       v0.1.1     v0.1.1    No
racer           v1.2.10    v1.2.10   No
rustfmt         v0.6.3     v0.6.3    No
treesize        v0.2.1     v0.2.1    No

Updating clippy
    Updating registry `https://github.com/rust-lang/crates.io-index`
    [...]
error: failed to compile `clippy v0.0.103`, intermediate artifacts can be found at `T:\-_-TEM~1\cargo-install.BlWxNojuYwrd`

Caused by:
  Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.

Updated 0 packages.
Failed to update clippy.
```

Will release in a bit

**Comment by nabijaczleweli** [comment]
Released in [`v0.6.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v0.6.0).
